lib/rules/constructor-super.js

```diff
@@ -31,6 +25,57 @@ function isConstructorFunction(node) {
     );
 }
 
+/**
+ * Checks whether a given node can be a constructor or not.
+ *
+ * @param {ASTNode} node - A node to check.
+ * @returns {boolean} `true` if the node can be a constructor.
+ */
+function isPossibleConstructor(node) {
+    if (!node) {
+        return false;
+    }
+
+    switch (node.type) {
+        case "ClassExpression":
+        case "FunctionExpression":
+        case "ThisExpression":
+        case "MemberExpression":
+        case "CallExpression":
+        case "NewExpression":
+        case "YieldExpression":
+        case "TaggedTemplateExpression":
+        case "MetaProperty":
+            return true;
+
+        case "Identifier":
+            return node.name !== "undefined";
```

Comment:
`undefined` can be shadowed. You either shouldn't make this exception, or you should make this exception for *every* non-constructor intrinsic (under the assumption that none of them could be shadowed). I would go with the former. It is safer.